cql3/Cql.g

```diff
@@ -415,11 +430,11 @@ selectClause returns [std::vector<shared_ptr<raw_selector>> expr]
 
 selector returns [shared_ptr<raw_selector> s]
     @init{ shared_ptr<cql3::column_identifier> alias; }
-    : us=unaliasedSelector (K_AS c=ident { alias = c; })? { $s = ::make_shared<raw_selector>(us, alias); }
+    : us=unaliasedSelector (K_AS c=ident { alias = c; })? { $s = ::make_shared<raw_selector>(std::move(us), alias); }
     ;
 
-unaliasedSelector returns [expression s]
-    @init { expression tmp; }
+unaliasedSelector returns [uexpression s]
```

Comment:
Looks good. By the way, another theoretical benefit of this "uninitialized expession" thing is that it allows (I think) less stack memory use during recursive parsing. In Alternator, I could barely fit "((((...(((" (4000 parentheses) on the stack because of the large stack use of every recursive parser call. Maybe with the "uninitialized" trick the stack usage can be reduced.

Reply:
I don't think it helps with the stack allocation. These uexpressions are eventually all initialized (or an exception is thrown).

Reply:
In fact it increases stack usage (from 8 bytes to 16).

Reply:
Oh, I misunderstood the uexpression implementation to be a unique_ptr (8 bytes) and the original "expression" to be something big and complicated...